acra/acra-in-depth/security-design/_index.md

```diff
@@ -2,3 +2,165 @@
 title: Security design
 bookCollapseSection: true
 ---
+
+## Security design
+
+Acra is built to provide selective encryption only to the records that require such protection.
+AcraWriter provides the app developer with the means to encrypt these records via wrapping anything in a function that outputs [AcraStruct]({{< ref "acra/acra-in-depth/data-structures/#understanding-acrastruct" >}}), a crypto container decryptable by AcraServer, which is then stored in a database.
+
+Any secret contained within an app (i.e. passwords in config files) can be leaked. There are many reasons why a developer might not want to use the user password as a secret - not all systems have to be end-to-end encrypted, even though we'd like it to be that way. An example of a good trust model is to use asymmetric cryptography to "send" this data (for the encryption of which we only have a public key) to some remote party. A leakage of this public key will still be insufficient for decrypting anything from the database.
+
+### Threat model
+
+We expect that anything, apart from AcraServer (which must remain uncompromised), can be compromised and that any piece of data can leak outside the system.
+
+Our goal is to provide the two following guarantees:
+
+**Guarantee 1**: If the attacker compromises the system through collecting all the data from any (or all) components other than AcraServer, and AcraServer remains uncompromised, this data will not be sufficient to decrypt protected entries.
+
+More specifically, AcraServer's key storage is the 'crown jewel' of the whole security scheme and should be protected with the typical server security measures.
```

Comment:
> with the typical server security measures.

This reads like you can run everything else with no security thought at all (= not ”typical measures”), `root/root` and public open ports, and still end up totally fine. I don't think Acra works that way (yet?)

Instead, I think you should make extra effort to secure the AcraServer and its keys.

Reply:
There is the main focus on keys in context of security. Same is true for PKI and root keys. Your environment that uses PKI should be protected too. But the core of the security schema is the root keys used for PKI that open access for whole environment, not for one specific node/app/host.

Reply:
I think, @ilammy 's point targets particularly wording "typical security measures" which would be better to rephrase like "special security measures"